src/file.cpp

```diff
@@ -1111,6 +1149,30 @@ namespace libtorrent
 		remove(f, ec);
 	}
 
+	void remove_empty(std::string const& f, error_code& ec)
+	{
+		ec.clear();
+
+		file_status s;
+		stat_file(f, &s, ec);
+		if (ec) return;
+
+		if (s.mode & file_status::directory)
+		{
+			for (directory i(f, ec); !i.done(); i.next(ec))
+			{
+				if (ec) return;
+
+				std::string p = i.file();
+				if (p == "." || p == "..") continue;
+				remove_empty(combine_path(f, p), ec);
+				ec.clear();
+			}
+
+			remove(f, ec);
```

Comment:
it seems like this function relies on the operating system to fail when trying to delete a directory that isn't empty. This would fail with an error along the lines "directory not empty", which I imagine would propagate to the caller, despite _some_ directories presumably being removed. maybe the name should suggest that it's just referring to directories (not empty files for instance).

would it make sense to return an indication to the caller that the directory was not empty perhaps? and once that has been received, it will be returned to its caller etc.

Also, removing all empty directories indiscriminately seems a bit strange. shouldn't it at least just consider the ones belonging to a specific torrent?